### PR TITLE
Heretic Influences use alt appearances (so ghosts can now see them (semi-transparent))

### LINKED ISSUE #77530: Cultist Keeping Fellow cultist icons after being unconverted.
## Round ID:[212250](https://scrubby.melonmesa.com/round/212250)

I spawned roundstart cultist did some cult shit got robusted husked by fire, then unhusked and revived while being fed holy water
stayed in crit for a while on a stasis bed then succumbed right before i died because they where not healing me, got fed a fuck ton more holy water, woke up out of crit after getting cpr said some shit on cult comms got fed a fuck ton MORE holy water, then got fed a holymelon that unculted me but left me with the little icons next to peoples heads (example picture below) went on to shoot the QM because i though i was in the cult still, leading to the bwoinkaning.
image

## Testmerges:

Dont think it was caused by any test merges
## Reproduction:

Spawn as roundstart cultist, die to fire and get husked, fed holy water the whole time revived and then fed more holy water then a holymelon

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Removes the rather overcomplicated system behind heretic influences and makes them use alt appearances

Fixes a few bugs involving alt appearances that I noticed in making them

I made this PR 7 months ago and forgot to Pr it

## Why It's Good For The Game

Less complex codes and lets observers get a cut in on the action.

Should fix #77530

## Changelog

:cl: Melbert
refactor: Refactored heretic influences a tiny bit, now ghosts can see them! Report any oddities.
/:cl:

### CODE REVIEW

**Comment by YesterdaysPromise** [comment]
abusable for both heretics and validhunters by just getting yourself killed and revived.

**Comment by SmArtKar** [comment]
Having some degree of separation between true observers and dead people would actually be really useful for features like these

**Comment by MrMelbert** [comment]
> Abusable for both heretics and validhunters by just getting yourself killed and revived. Hope its actually _just_ observers, people who join as ghosts, rather than any and all ghosts?

then they just get banned? same as if you died and looked for invisible cult runes?

**Comment by MrMelbert** [comment]
> Having some degree of separation between true observers and dead people would actually be really useful for features like these

you're free to add it afterwards it's an existing feature

**Comment by CCC23** [comment]
killing myself as heretic to avoid using the security cameras to look for influences

File: code/game/atom/alternate_appearance.dm [hunk 1]
```diff
@@ -35,20 +35,53 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
...
+
+/datum/atom_hud/alternate_appearance/hide_from(mob/former_viewer, absolute)
+	. = ..()
+	if(!former_viewer || hud_atoms_all_z_levels[former_viewer] >= 1)
+		return
+	untrack_mob(former_viewer)
+
+/// Unregisters the signals that were tracking the mob's state
+/datum/atom_hud/alternate_appearance/proc/untrack_mob(mob/former_viewer)
+	SHOULD_CALL_PARENT(TRUE)
```
Comment (Time-Green) [inline code/game/atom/alternate_appearance.dm:75]: Ditto

File: code/game/atom/alternate_appearance.dm [hunk 2]
```diff
@@ -99,6 +132,24 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
...
 		QDEL_NULL(ghost_appearance)
 
+/datum/atom_hud/alternate_appearance/basic/track_mob(mob/new_viewer)
+	. = ..()
+	RegisterSignals(new_viewer, list(
+		COMSIG_MOB_ANTAGONIST_REMOVED,
+		COMSIG_MOB_GHOSTIZED,
+		COMSIG_MOB_MIND_TRANSFERRED_INTO,
+		COMSIG_MOB_MIND_TRANSFERRED_OUT_OF,
+	), PROC_REF(check_hud), override = TRUE)
```
Comment (Time-Green) [inline code/game/atom/alternate_appearance.dm:142]: I get why you used the other override, but why this one? Shouldn't each alternate_appearance instance not be tracking duplicate mobs?

**Review by Time-Green — CHANGES_REQUESTED** [review]

**Comment by nikothedude** [comment]
> Abusable for both heretics and validhunters by just getting yourself killed and revived. Hope its actually _just_ observers, people who join as ghosts, rather than any and all ghosts?

you can literally stalk every single person in the game at any time by dying and observing into all their inventories we are so far past the point of this being an issue

File: code/game/atom/alternate_appearance.dm (hunk 2 above)
Comment (MrMelbert) [reply code/game/atom/alternate_appearance.dm:139]: Showing a hud to the same mod twice is valid

**Pull request merged**